frontend/app/services/wsaddress-service.server.ts

```diff
@@ -0,0 +1,151 @@
+  warnings: z.string().optional(),
+});
+
+
+export const getWSAddressService = moize.promise(createWSAddressService, { onCacheAdd: () => log.info('Creating new WSAddress service') });
+
+async function createWSAddressService() {
+  const { INTEROP_API_BASE_URI } = getEnv();
+
+  async function parseWSAddress({addressLine, city, province, postalCode, country, language}: ParseWSAddressRequestDTOProps) {
+    const url = `${INTEROP_API_BASE_URI}/address/parse`;
+    const response = await fetch(url, {
+      method: 'POST',
+      body: JSON.stringify(new ParseWSAddressRequestDTO(addressLine, city, province, postalCode, country, language)),
+    });
+
+    if (response.ok) {
+      const parseData = ParseWSAddressResponseSchema.parse(await response.json());
+      if(parseData.statusCode === 'Valid') return new ParseWSAddressResponseDTO(
+        parseData.responseType,
+        parseData.addressLine,
+        parseData.city,
+        parseData.province,
+        parseData.postalCode,
+        parseData.streetNumberSuffix,
+        parseData.streetDirection,
+        parseData.unitType,
+        parseData.unitNumber,
+        parseData.serviceAreaName,
+        parseData.serviceAreaType,
+        parseData.serviceAreaQualifier,
+        parseData.cityLong,
+        parseData.cityShort,
+        parseData.deliveryInformation,
+        parseData.extraInformation,
+        parseData.statusCode,
+        parseData.canadaPostInformation,
+        parseData.message,
+        parseData.addressType,
+        parseData.streetNumber,
+        parseData.streetName,
+        parseData.streetType,
+        parseData.serviceType,
+        parseData.serviceNumber,
+        parseData.country,
+        parseData.warnings,
+        parseData.functionalMessages,
+      );
+    }
+
+    log.error('%j', {
+      message: 'Failed to parse the address',
+      status: response.status,
+      statusText: response.statusText,
+      url: url,
+      responseBody: await response.text(),
+    });
+
+    throw new Error(`Failed to parse the address. Status: ${response.status}, Status Text: ${response.statusText}`);
+  }
+
+  async function validateWSAddress({addressLine, city, province, postalCode, country, geographicScope, parseType, language}: ValidateWSAddressRequestDTOProps) {
+    const url = `${INTEROP_API_BASE_URI}/address/validate`;
+    const response = await fetch(url, {
+      method: 'POST',
+      body: JSON.stringify(new ValidateWSAddressRequestDTO(addressLine, city, province, postalCode, country, geographicScope, parseType, language)),
+    });
+
+    if (response.ok) {
+      const validationData = ValidateWSAddressResponseSchema.parse(await response.json());
+
+      if (validationData.statusCode === 'Valid') return new ValidateWSAddressResponseDTO(
+        validationData.responseType,
+        validationData.statusCode,
+        validationData.functionalMessages,
+        validationData.message,
+        validationData.warnings
+      );
+    }
```

Comment:
See my other comments on removing the class implementation of this.

Reply:
Moved response DTO to wsaddress-service and modified the return